src/Analysis/Ast/Impl/Specializations/Typing/Types/NamedTupleType.cs

```diff
@@ -27,20 +27,25 @@ internal sealed class NamedTupleType : TypingTupleType, ITypingNamedTupleType {
         /// <summary>
         /// Creates type info for a strongly-typed tuple, such as Tuple[T1, T2, ...].
         /// </summary>
-        public NamedTupleType(string tupleName, IReadOnlyList<string> itemNames, IReadOnlyList<IPythonType> itemTypes, IPythonInterpreter interpreter)
-            : base(itemTypes, interpreter) {
+        public NamedTupleType(string tupleName, IReadOnlyList<string> itemNames, IReadOnlyList<IPythonType> itemTypes, IPythonModule declaringModule, IPythonInterpreter interpreter)
```

Comment:
When type is declared via `NamedTuple`, if should belong to the module it is declared in and not to `typing`, similar to how other generics behave when specific type is created.